operator/pkg/name/name.go

```diff
@@ -50,7 +50,11 @@ const (
 	EgressComponentName  ComponentName = "EgressGateways"
 
 	// Addon components
-	AddonComponentName ComponentName = "Addon"
```

Comment:
we can change it to "AddonComponents" instead? it would be used at multiple places

Reply:
Not sure what you mean? The idea of this PR was to remove the bundling of addonComponents into a single entity; I think they should be handled at the same level of abstraction as normal components, otherwise we duplicate logic everywhere